keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -355,6 +423,193 @@ class ClusterHandler(
     )
   }
 
+  /**
+   * @return list of stages to remove or create scaling policies in-place on the
+   * current serverGroup.
+   *
+   * Scaling policies are treated as immutable by keel once applied. If an existing
+   * policy is modified, it will be deleted and reapplied via a single task.
+   */
+  private fun ResourceDiff<ServerGroup>.modifyScalingPolicyJob(startingRefId: Int = 0): List<Map<String, Any?>> {
+    var (refId, stages) = toDeletePolicyJob(startingRefId)
+    val newTargetPolicies = when (current) {
+      null -> desired.scaling.targetTrackingPolicies
+      else -> desired.scaling.targetTrackingPolicies
+        .subtract(current!!.scaling.targetTrackingPolicies)
+    }
+    val newStepPolicies = when (current) {
+      null -> desired.scaling.stepScalingPolicies
+      else -> desired.scaling.stepScalingPolicies
+        .subtract(current!!.scaling.stepScalingPolicies)
+    }
+
+    if (newTargetPolicies.isNotEmpty()) {
+      val (newRef, jobs) = newTargetPolicies.toCreateJob(refId, current ?: desired)
+      refId = newRef
+      stages.addAll(jobs)
+    }
+
+    if (newStepPolicies.isNotEmpty()) {
+      stages.addAll(newStepPolicies.toCreateJob(refId, current ?: desired))
+    }
+
+    return stages
+  }
+
+  private fun ResourceDiff<ServerGroup>.toDeletePolicyJob(startingRefId: Int):
+    Pair<Int, MutableList<Map<String, Any?>>> {
+    var refId = startingRefId
+    val stages: MutableList<Map<String, Any?>> = mutableListOf()
+    if (current == null) {
+      return Pair(refId, stages)
+    }
+    val current = current!!
+    val targetPoliciesToRemove = current.scaling.targetTrackingPolicies.filterNot {
+      desired.scaling.targetTrackingPolicies.contains(it)
+    }
+    val stepPoliciesToRemove = current.scaling.stepScalingPolicies.filterNot {
+      desired.scaling.stepScalingPolicies.contains(it)
+    }
+    val policyNamesToRemove = targetPoliciesToRemove.mapNotNull { it.name } +
+      stepPoliciesToRemove.mapNotNull { it.name }
+        .toSet()
+
+    stages.addAll(
+      policyNamesToRemove
+        .map {
+          refId++
+          mapOf(
+            "refId" to refId.toString(),
+            "requisiteStageRefIds" to when (refId) {
+              0, 1 -> listOf()
+              else -> listOf((refId - 1).toString())
+            },
+            "type" to "deleteScalingPolicy",
+            "policyName" to it,
+            "cloudProvider" to CLOUD_PROVIDER,
+            "credentials" to desired.location.account,
+            "moniker" to mapOf(
```

Comment:
can this just be `"moniker" to current.moniker`?

Reply:
🤦‍♂ 💯

Reply:
Oh, this is actually done in the pre-existing orca task generating methods in keel too. It looks like keel's `Moniker` class doesn't quite serialize to orca's representation of a cluster moniker; in this case the cluster's `moniker.name` is mapped to `moniker.cluster` for orca.

Reply:
Looks like a good candidate for a `toOrcaMoniker()` or something in our `Moniker` class...